### ISSUE TITLE
Example code triggers assertion failure on Mac OSX

### ISSUE THREAD

**Issue by Geod24** [aferust/opencvd#9]
System:
- Mac OSX Cataline 10.15.3
- DMD 2.090.1 / LDC 1.20.0
- Apple clang version 11.0.0 (clang-1100.0.33.17)
- OpenCV 4.2.0 built via Homebrew:
```
General configuration for OpenCV 4.2.0 =====================================
  Version control:               unknown

  Extra modules:
    Location (extra):            /tmp/opencv-20200105-17262-cwpzm4/opencv-4.2.0/opencv_contrib/modules
    Version control (extra):     unknown

  Platform:
    Timestamp:                   2020-01-05T21:53:54Z
    Host:                        Darwin 19.2.0 x86_64
    CMake:                       3.16.2
    CMake generator:             Unix Makefiles
    CMake build tool:            /usr/local/Homebrew/Library/Homebrew/shims/mac/super/gmake
    Configuration:               Release

  CPU/HW features:
    Baseline:                    SSE SSE2 SSE3 SSSE3 SSE4_1 POPCNT SSE4_2
      requested:                 DETECT
      disabled:                  AVX AVX2
    Dispatched code generation:  FP16 AVX AVX2 AVX512_SKX
      requested:                 SSE4_1 SSE4_2 AVX FP16 AVX2 AVX512_SKX
      FP16 (0 files):            + FP16 AVX
      AVX (4 files):             + AVX
      AVX2 (27 files):           + FP16 FMA3 AVX AVX2
      AVX512_SKX (3 files):      + FP16 FMA3 AVX AVX2 AVX_512F AVX512_COMMON AVX512_SKX

  C/C++:
    Built as dynamic libs?:      YES
    C++ Compiler:                /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++  (ver 11.0.0.11000033)
    C++ flags (Release):         -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include/darwin   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG  -DNDEBUG
    C++ flags (Debug):           -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include/darwin   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
    C Compiler:                  /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang
    C flags (Release):           -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include/darwin   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG  -DNDEBUG
    C flags (Debug):             -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include/darwin   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
    Linker flags (Release):      -Wl,-dead_strip
    Linker flags (Debug):        -Wl,-dead_strip
    ccache:                      NO
    Precompiled headers:         NO
    Extra dependencies:
    3rdparty dependencies:

  OpenCV modules:
    To be built:                 aruco bgsegm bioinspired calib3d ccalib core datasets dnn dnn_objdetect dnn_superres dpm face features2d flann freetype fuzzy gapi hfs highgui img_hash imgcodecs imgproc line_descriptor ml objdetect optflow phase_unwrapping photo plot python3 quality reg rgbd saliency sfm shape stereo stitching structured_light superres surface_matching text tracking video videoio videostab xfeatures2d ximgproc xobjdetect xphoto
    Disabled:                    hdf python2 world
    Disabled by dependency:      -
    Unavailable:                 cnn_3dobj cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev cvv java js matlab ovis ts viz
    Applications:                apps
    Documentation:               NO
    Non-free algorithms:         YES

  GUI:
    Cocoa:                       YES

  Media I/O:
    ZLib:                        /usr/lib/libz.dylib (ver 1.2.11)
    JPEG:                        build-libjpeg-turbo (ver 2.0.2-62)
    WEBP:                        build (ver encoder: 0x020e)
    PNG:                         /usr/local/lib/libpng.dylib (ver 1.6.37)
    TIFF:                        /usr/local/lib/libtiff.dylib (ver 42 / 4.1.0)
    OpenEXR:                     /usr/local/lib/libImath-2_4.dylib /usr/local/lib/libIlmImf-2_4.dylib /usr/local/lib/libIex-2_4.dylib /usr/local/lib/libHalf-2_4.dylib /usr/local/lib/libIlmThread-2_4.dylib (ver 2_4)
    HDR:                         YES
    SUNRASTER:                   YES
    PXM:                         YES
    PFM:                         YES

  Video I/O:
    FFMPEG:                      YES
      avcodec:                   YES (58.54.100)
      avformat:                  YES (58.29.100)
      avutil:                    YES (56.31.100)
      swscale:                   YES (5.5.100)
      avresample:                YES (4.0.0)
    AVFoundation:                YES

  Parallel framework:            TBB (ver 2020.0 interface 11100)

  Trace:                         YES (with Intel ITT)

  Other third-party libraries:
    Intel IPP:                   2019.0.0 Gold [2019.0.0]
           at:                   /tmp/opencv-20200105-17262-cwpzm4/opencv-4.2.0/build/3rdparty/ippicv/ippicv_mac/icv
    Intel IPP IW:                sources (2019.0.0)
              at:                /tmp/opencv-20200105-17262-cwpzm4/opencv-4.2.0/build/3rdparty/ippicv/ippicv_mac/iw
    Lapack:                      YES (/usr/local/opt/openblas/lib/libopenblas.dylib)
    Eigen:                       YES (ver 3.3.7)
    Custom HAL:                  NO
    Protobuf:                    build (3.5.1)

  OpenCL:                        YES (no extra features)
    Include path:                NO
    Link libraries:              -framework OpenCL

  Python 3:
    Interpreter:                 /usr/local/opt/python/bin/python3 (ver 3.7.6)
    Libraries:                   /usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/config-3.7m-darwin/libpython3.7.dylib (ver 3.7.6)
    numpy:                       /usr/local/lib/python3.7/site-packages/numpy/core/include (ver 1.17.4)
    install path:                lib/python3.7/site-packages/cv2/python-3.7

  Python (for build):            /usr/bin/python2.7

  Java:
    ant:                         NO
    JNI:                         /Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include /Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include/darwin /Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.1.jdk/Contents/Home/include
    Java wrappers:               NO
    Java tests:                  NO

  Install to:                    /usr/local/Cellar/opencv/4.2.0_1
-----------------------------------------------------------------
```

Using cf49b221cbf01f77a478a1de16a977ba029d4658 cloned in `submodules/opencvd`

```json
    "dependencies": {
        "opencvd": { "path": "submodules/opencvd" }
    },
    "lflags": [ "-L." ],
    "libs": [
        "opencv4",
        "opencvcapi",
        "opencvcapi_contrib"
    ]
```

Running the example code triggers:
```
Copying files for opencvd...
Running ./motion
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: OpenCV(4.2.0) /tmp/opencv-20200105-17262-cwpzm4/opencv-4.2.0/modules/core/src/matrix.cpp:406: error: (-215:Assertion failed) m.dims >= 2 in function 'Mat'

Program exited with code -6
```

**Comment by aferust** [comment]
Which example is that?
Please provide a very brief example allowing me to reproduce the error. I don't access a osx right now though.

**Comment by Geod24** [comment]
The example linked here: https://github.com/aferust/opencvd/tree/cf49b221cbf01f77a478a1de16a977ba029d4658#some-examples-to-show-how-it-looks-like

**Comment by aferust** [comment]
Could you please make below changes and confirm if the problem is solved?

Replace this:
auto mt = Mat(20, 20, CV_8U);
with
auto mt = Mat(20, 20, CV_8UC3);

**Comment by aferust** [comment]
I assume this issue is solved after fixing a bug in the example code 65f68fa. It is interesting that the buggy code was running on Windows.
